include/nbl/builtin/hlsl/cmath.hlsl

```diff
@@ -0,0 +1,258 @@
+// Copyright (C) 2022 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+#ifndef _NBL_BUILTIN_HLSL_CMATH_INCLUDED_
+#define _NBL_BUILTIN_HLSL_CMATH_INCLUDED_
+
+#include <nbl/builtin/hlsl/limits.hlsl>
+
+// C++ headers
+#ifndef __HLSL_VERSION
+#include <cmath>
+#endif
+
+namespace nbl
+{
+namespace hlsl
+{
+
+
+#ifdef __cplusplus
```

Comment:
don't use `__cplusplus`, use `ifndef __HLSL_VERSION` (there might be a C++-like compiler emitting SPIR-V soon - either Shady or Clang-HLSL)